ambry-network/src/test/java/com.github.ambry.network/SSLSelectorTest.java

```diff
@@ -203,6 +202,19 @@ public void testEmptyRequest()
     assertEquals("", blockingRequest(connectionId, ""));
   }
 
+  @Test
+  public void testSSLConnect()
+      throws IOException {
+    String connectionId =
+        selector.connect(new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE, PortType.SSL);
+    Assert.assertFalse("Channel should not be ready by now (until handshake completes)",
+        selector.isChannelReady(connectionId));
+    while (!selector.connected().contains(connectionId)) {
+      selector.poll(10000L);
+    }
+    Assert.assertTrue("Channel should have been ready by now ", selector.isChannelReady(connectionId));
+  }
+
   private String blockingRequest(String connectionId, String s)
```

Comment:
Is it possible to add a test case for handshake failure and cleanup part? Basically to cover all new code.